src/actions/guides/http_and_routing/routing_and_params.cr

```diff
@@ -16,11 +16,13 @@ class Guides::HttpAndRouting::RoutingAndParams < GuideAction
 
     ## Routing
 
-    Unlike many frameworks, there is no separate routes file. An action declares
-    which route it handles in the class itself.
-
-    You can route to an action by using `get`, `put`, `post`, `patch`, `trace`, and `delete` macros, or
-    [Lucky can figure it out for you](##{ANCHOR_AUTOMATICALLY_GENERATE_RESTFUL_ROUTES}).
+    Unlike many frameworks, there is no separate route definition file.
+    
+    The name of an action class defines the route that triggers that action by default.
+    Lucky automatically configures [RESTful routes (see below)](##{ANCHOR_AUTOMATICALLY_GENERATE_RESTFUL_ROUTES}).
+    
+    For example, an action class `Item::Show` defaults to handle `get "/item/:item_id"`, a HTTP GET request to show a specific item.
+    The route can also be defined for specific request types by using `get`, `put`, `post`, `patch`, `trace`, and `delete` macros.
```

Comment:
We may actually remove this functionality https://github.com/luckyframework/lucky/issues/789

So I think we should leave this as it was since we don't want over  emphasize something that is likely to be removed

Reply:
Too bad, I liked that short syntax. What about adding 

```suggestion
    Lucky started out automatically configuring [RESTful routes (see below)](##{ANCHOR_AUTOMATICALLY_GENERATE_RESTFUL_ROUTES}), based on the class name.
    For example, an action class `Item::Show` defaults to handle `get "/item/:item_id"`,
    a HTTP GET request to show a specific item. However, removal of this feature
    is [discussed](https://github.com/luckyframework/lucky/issues/789). 
    
    Explicit routes can be defined for specific request types by using the `get`, `put`, `post`, `patch`, `trace`, and `delete` macros.
```

Reply:
Since this will go live right after I merge I think that may be confusing since you currently can use the `route` macro.

The plan is that the generators will still be able to infer the route name for you so it can still be fast to add new actions. The other reason for this change is that we want to allow specifying path types, which is hard/impossible/weird with the `route` macro:

```crystal
# Will make sure the id can be parsed to a User::ID and can only be used in User queries
get "/users/:user_id(User::ID)" do
end
```

I hope that kind of explains the rationale. If we find a way to bring it back though maybe we will! We're still experimenting with the ideal APIs :)

Reply:
could route accept a class when building
route Show => User::ID

```suggestion
    Lucky can automatically infer [RESTful routes (see below)](##{ANCHOR_AUTOMATICALLY_GENERATE_RESTFUL_ROUTES}), based on the name of the class.
    For example, an action named `Item::Show` defaults to handle `get "/item/:item_id"`,
    a HTTP GET request to show a specific item.
    Executing `lucky gen.action Item::Show` will create the action with this default route.
    The `route` `nested-route` macros do also still automatically create these default route paths,
    however, their depreciation is [discussed](https://github.com/luckyframework/lucky/issues/789). 
    
    Explicit routes can be defined for specific request types by using the `get`, `put`, `post`, `patch`, `trace`, and `delete` macros.
```
Not sure, do you mean something like this:
```
class Users::Show < BrowserAction
  match_param_to[User::ID]
  route
    text "Now #{user_id} is ensured to be a known user."
  end
end
```

Reply:
But what if there are multiple params?

```crystal
class Users::Projects::Show < BrowserAction
  match_param user_id : User::ID
  match_param project_id : Project::Id

  route
    text "Now #{user_id} is ensured to be a known user."
  end
end
```

The problem here is it is much more verbose and requires inner knowledge of how Lucky is generating params. Whereas `get "/users/:user_id(User::ID)/projects/:project_id(Project::ID)"` is not much typing and makes a lot of sense just from looking at it. At least that's my opinion :P But if you have ideas please leave them on that issues so we can keep track of everything

Reply:
BTW what was the reason to drop `action`?
To me, `route` lets me expect path definition stuff to be in the block, wheras that context needs to be thought looking at the class name, wheras looking at the code block it is all about the action.

Reply:
Think: `action` determines the default routing path automatically, implicitly.

Reply:
People were confused by `action` almost universally. `route` clicked a bit better it seemed. But I think it is subjective and depends on what you're used to or expect 🤷‍♂️ So far I haven't heard complains about the `route` name

Reply:
Yeah, I think what was missing was just an explanation to understand how in Lucky the object focus is on the "action class", instead of "routes and controller files".

When the intro properly explains it (is easy to understand for new and experienced users) I think it could have actually been better to use `action` instead of `route` to let the focus stay on the action name.

But with dropping the `route` macro as well it will be mood of course, and `get` et. al. are clearer for the long run readability as you were able to convnice me.